### PR TITLE
PROD-740 #2921 - Fix: New User Role Assignment doesn't stick upon initial User creation

### LINKED ISSUE #2921: New User Role Assignment doesn't stick upon initial User creation
**Describe the bug**
The New user Role-Assignments not being honored by the WP Backend System. It is not a third-party plugin or theme issue. The steps/actions that led to this issue are listed below. I have taken time and effort to distill the steps down to as few as possible.

**To Reproduce**
1. Given a basic BuddyBoss setup—with both mandatory plugins and theme
active—add a couple of basic, manually-created test users for good measure,
as well as to test and see (first-hand) that the system works as expected.

2. Go to BuddyBoss > Components and enable all components

3. Go to BuddyBoss > Settings > Profiles > Profile Types

   - a. select “Enable profile types to give members unique profile fields and
permissions”

   - b. click [Save Settings]

4. Go to BuddyBoss > Settings > Groups > Group Types

   - a. select “Enable group types to better organize groups enabled”

   - b. click [Save Settings]

5. Add a couple of interim/test Profile Types

   - a. BuddyBoss > Profile > Profile Types [Add New]

   - b. Model your test Profile Types according to the ScreenShot ()

   - c. Click [Publish]

6. Go to BuddyBoss> Settings > Profiles > Profile Types:

   - a. From the Default Profile Type Dropdown, select “ProfileType-01s”*

   - b. Click [Save Settings]*

7. Create a new member by going to: WP > Users > Add New

   - a. Enter desired test user data

   - b. select any role other than Subscriber”

   - c. click [Add New User]

**Screenshots**
https://www.youtube.com/watch?v=84soWG78UZM

**Support ticket links**
https://secure.helpscout.net/conversation/1618716025/161848?folderId=3709082

Issue: #2140 

**Jira issue** : [PROD-740]

[PROD-740]: https://buddyboss.atlassian.net/browse/PROD-740?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by milanhirapra**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2921 .

### How to test the changes in this Pull Request:

1. Activate plugins and theme 
2. Add a couple of basic, manually-created test users for good measure
3. Go to BuddyBoss > Components and enable all components
4. Go to BuddyBoss > Settings > Profiles > Profile Types
5. select “Enable profile types to give members unique profile fields and permissions” and update it
6. Go to BuddyBoss > Settings > Groups > Group Types
7. select “Enable group types to better organize groups enabled” and update it
8. Add a couple of Profile Types from BuddyBoss > Profile > Profile Types [Add New]
9. Go to BuddyBoss> Settings > Profiles > Profile Types
10. From the **Default Profile Type** Dropdown, select any one Profile Type from the list and save it

**Scenario 1**

1. Create a new member by going to WP > Users > Add New
2. Enter desired test user data
3. Select any role other than **Subscriber** and create it.
4. Go to WP > Users and find this user and see that the role is correct.

**Scenario 2**

1. Again Add a couple of Profile Types from BuddyBoss > Profile > Profile Types [Add New]
2. Go to the website with the admin user
3. Click on send invite button ( Go profile image in the upper right corner > Email Invites > Send Invites )
4. Send Invite - make sure to choose a profile type
5. Go to email and click on the link to register
6. Activate account after register
7. Go to WP > Users and find registered user and see that the profile type is correct.

### Proof Screenshots or Video
https://vimeo.com/617965408/edc1577e27

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by dipskakadiya — APPROVED** [review]

File: src/bp-members/bp-members-functions.php [hunk 1]
```diff
@@ -4403,61 +4417,40 @@ function bp_assign_default_member_type_to_activate_user_on_admin( $user_id ) {
...
-                $bp_user = new WP_User( $user_id );
-                foreach ( $bp_user->roles as $role ) {
-                    // Remove role
-                    $bp_user->remove_role( $role );
-                }
-                // Add role
-                $bp_user->add_role( $selected_member_type_wp_roles[0] );
-            }
-        }
+				} else {
```
Comment (dipskakadiya) [inline src/bp-members/bp-members-functions.php:4420]: @milanhirapra Can you run PHPCS to fix formatting?

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

**Review by dipskakadiya — APPROVED** [review]

**Comment by mckdlrs** [comment]
Retested in the QA ongoing branch
Status: Not yet fixed

Proof Screenshots or Video:
https://www.loom.com/share/6ab1f38b3375498d9c8f93fdad6d99e8

**Review by chetansatasiya — DISMISSED** [review]

**Comment by basil-buddyboss** [comment]
Hello @milanhirapra 

Both scenarios have been tested in the QA-ongoing branch and the issue is now resolved.

**Pre-condition:**
1. All BB platform component is enabled.
2. Profile type and Group Type is enabled
3. Create a couple of profile types
4. Select a default profile type to be auto-assigned to users during registration

**To reproduce:
Scenario 1:**
1. Create a new member from the WP> Users> Add new
2. Select any WP role other than the subscriber role.
3. Go to the WP user list and check if the WP role of the user is correct.

**Scenario 2:**
1. Create a new profile type and the WP role is admin
2. Set this profile type to a user
3. Go to the profile of that user and send an email invite - make sure to choose a profile type
4. Register and activate the account
5. View the WP user list and check if the profile type assigned to the user is correct.

Video:
**Scenario 1:**
https://watch.screencastify.com/v/q8Qh4qWJCRPJ5UmnigWx
**Scenario 2:**
https://watch.screencastify.com/v/9GppCJJ1WnVgJlNGkYY1

**Status: Passed**

**Comment by joeytoribio** [comment]
Hello @milanhirapra,

This has been tested in the qa-approved branch without any issues.

**Status: PASSED**

**Comment by sourabhmatolia** [comment]
**UPDATE**

This is not done correctly. I will share more details later. Thanks.

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**